tmpl/wiki/hello-world.md

```diff
@@ -1,232 +1,486 @@
+
 ### Step 1: Hello World!
 
-As a first step, let's build and run the MirageOS "Hello World" unikernel -- this
-will print `hello\nworld\n` 4 times before terminating:
+As a first step, let's build and run the MirageOS "Hello World" unikernel --
+this will print a log message with the word `hello` 4 times before terminating:
 
 ```
-hello
-world
-hello
-world
-hello
-world
-hello
-world
+2017-02-08 09:54:44 -01:00: INF [application] hello
+2017-02-08 09:54:45 -01:00: INF [application] hello
+2017-02-08 09:54:46 -01:00: INF [application] hello
+2017-02-08 09:54:47 -01:00: INF [application] hello
 ```
 
 First, let's look at the code:
 
 ```
-$ cat console/unikernel.ml
 open Lwt.Infix
 
-module Main (C: V1_LWT.CONSOLE) = struct
+module Hello (Time : Mirage_time_lwt.S) = struct
+
+  let start pclock =
 
-  let start c =
     let rec loop = function
       | 0 -> Lwt.return_unit
       | n ->
-        C.log_s c "hello" >>= fun () ->
-        OS.Time.sleep 1.0 >>= fun () ->
-        C.log_s c "world" >>= fun () ->
+        Logs.info (fun f -> f "hello");
+        Time.sleep_ns (Duration.of_sec 1) >>= fun () ->
         loop (n-1)
     in
     loop 4
 
 end
 ```
 
-**Note**: If you aren't familiar with the Lwt library (and the `>>=` operator it provides),
-you may want to read at least the start of the [Lwt tutorial](tutorial-lwt) first.
-
 To veteran OCaml programmers among you, this might look a little odd: we have a
-`Main` module parameterised by another module (`C`, of type `CONSOLE`) that
-contains a method `start` taking a single parameter `c` (an instance of a
-`CONSOLE`). This is the basic structure required to make this a MirageOS
-unikernel rather than a standard OCaml POSIX application.
+`Main` module parameterised a module (`Time`, of type `Mirage_time_lwt.S`) that contains a method `start` taking an ignored parameter `_time` (an instance of a `time`).  This is the basic structure required to make this a MirageOS unikernel
+rather than a standard OCaml POSIX application.
+
+The module type for our `Time` module, `Mirage_time_lwt.S`, is defined in an
+external package [mirage-time](https://github.com/mirage/mirage-time).  The name `S` for "the module type of things like this" is a common OCaml convention (comparable to naming the most-used type in a module `t`).  There are many packages defining module types for use in Mirage.  For ease of discovery, a list of the module types that Mirage knows about is maintained
+in the [`types/`](https://github.com/mirage/mirage/tree/master/types) directory
+of the main MirageOS repository.  The `Mirage_types` module gives abstract definitions that leave some important primitives unspecified; the `Mirage_types_lwt` module contains more concrete definitions for use in programs.  Since you'll find yourself referring back to
+these quite often when building MirageOS applications, it's worth bookmarking
+the [documentation](http://mirage.github.io) for this module.
```

Comment:
this link should point to http://docs.mirage.io/mirage-types-lwt/Mirage_types_lwt/index.html !?